### DISCUSSION TITLE
Question on datasource uses

### DISCUSSION THREAD

**Discussion by gurpiarbassi** [flamingock/mongock#53] [Q&A] [answered]
This is a general question about using Mongock with Spring.
Is it possible to configure Mongock with spring to use a different user?

My use case is I have  a read-write user which I want my application to use and I want Mongock to use a different user which has more than just read-write i..e it will need permissions to create indexes, modify collections e.t.c

**Comment by dieppa** [comment, marked as answer]
Hello @gurpiarbassi ,


By default Mongock takes the main MongoTemplate injected when using `@EnableMongock`. We have in our roadmap a feature which allows you to decide which MongoTemplate bean to take, but it's not high priority.

However, you can do this with the builder approach by injecting a specific MongoTemplate for Mongock.

Something like this:


```java
public MongoTemplate mongockMongoTemplate(...) {
  return mongotemplate_with_other_permissions;
}

@Bean
public MongockInitializingBeanRunner mongockRunner(MongoTemplate mongockMongoTemplate,  ApplicationContext applicationContext) {
    return MongockSpringboot.builder()
           .setDriver(SpringDataMongoV3Driver.withDefaultLock(mongoTemplate))
           .setSpringContext(applicationContext)
            //...
           .buildInitializingBeanRunner();
}

```

I hope this helps 😃 

**Two more things**
 _if you want to keep update with new releases, follow us on [twitter](https://twitter.com/MongockTeam)_

_I move this to our discussion channel_
